### PR TITLE
add intersect()

### PR DESCRIPTION
**Pull request by thekuom**
Addresses #2064

I needed `INTERSECT` for one of my projects so I used the existing `UNION` as a model, since they are extremely similar. Thanks for the great library!

### CODE REVIEW

**Comment by thekuom** [comment]
Documentation PR: https://github.com/knex/documentation/pull/173

File: test/integration/builder/unions.js [hunk 1]
```diff
@@ -112,4 +112,120 @@ module.exports = function(knex) {
         );
     });
   });
+
+  if (
+    ['pg', 'mssql', 'pg-redshift', 'oracledb', 'sqlite3'].includes(
+      knex.client.driverName
+    )
+  ) {
+    describe('intersects', function() {
```
[the remark this answers is not in the document]
Comment (thekuom) [reply test/integration/builder/unions.js]: done

File: test/integration/builder/unions.js [hunk 2]
```diff
@@ -112,4 +114,220 @@ module.exports = function(knex) {
         );
     });
   });
+
+  if (
+    ['pg', 'mssql', 'pg-redshift', 'oracledb', 'sqlite3'].includes(
+      knex.client.driverName
+    )
+  ) {
+    describe.only('intersects', function() {
```
Comment (kibertoad) [inline test/integration/builder/unions.js]: .only shouldn't be committed

File: test/integration/builder/unions.js [hunk 3]
```diff
@@ -112,4 +114,220 @@ module.exports = function(knex) {
...
+            id: 5,
+            test_col_1: 1,
+            test_col_2: 2,
+            test_col_3: 1,
+          },
+        ]);
+      });
+
+      after(function() {
+        return knex.schema.dropTableIfExists('intersect_test');
```
Comment (kibertoad) [inline test/integration/builder/unions.js]: dropTableIfExists is unnecessary here, you can just drop it always.

File: test/unit/query/builder.js [hunk 4]
```diff
@@ -2190,6 +2190,330 @@ describe('QueryBuilder', function() {
     });
   });
 
+  it('intersects', function() {
+    var chain = qb()
```
Comment (kibertoad) [inline test/unit/query/builder.js]: why not const?

File: test/unit/query/builder.js [hunk 5]
```diff
@@ -2190,6 +2190,330 @@ describe('QueryBuilder', function() {
...
+        bindings: [1, 2],
+      },
+      sqlite3: {
+        sql:
+          'select * from `users` where `id` = ? intersect select * from `users` where `id` = ?',
+        bindings: [1, 2],
+      },
+    });
+
+    var multipleArgumentsChain = qb()
```
Comment (kibertoad) [inline test/unit/query/builder.js]: const?

File: types/knex.d.ts [hunk 6]
```diff
@@ -389,6 +392,19 @@ declare namespace Knex {
...
+    (
+      callback: QueryCallback | QueryBuilder | Raw,
+      wrap?: boolean
+    ): QueryBuilder;
+    (
+      callbacks: (QueryCallback | QueryBuilder | Raw)[],
+      wrap?: boolean
+    ): QueryBuilder;
+    (...callbacks: (QueryCallback | QueryBuilder | Raw)[]): QueryBuilder;
+    // (...callbacks: QueryCallback[], wrap?: boolean): QueryInterface;
```
Comment (kibertoad) [inline types/knex.d.ts]: should this be commented out?

File: test/integration/builder/unions.js (hunk 2 above)
Comment (thekuom) [reply test/integration/builder/unions.js]: Good call

File: test/integration/builder/unions.js (hunk 3 above)
Comment (thekuom) [reply test/integration/builder/unions.js]: Sure thing

File: test/unit/query/builder.js (hunk 4 above)
Comment (thekuom) [reply test/unit/query/builder.js]: No reason, will change

File: test/unit/query/builder.js (hunk 5 above)
Comment (thekuom) [reply test/unit/query/builder.js]: Will change

File: types/knex.d.ts (hunk 6 above)
Comment (thekuom) [reply types/knex.d.ts]: I'll remove it

**Comment by kibertoad** [comment]
Looks good, thank you!

**Pull request merged**